### PR TITLE
fix(schemas): fix the get interation/consent api bug

### PR DESCRIPTION
**Pull request by simeng-li**
### Reproduction Steps

- Create an organization scope with an empty description and assign this scope to a third-party application.

- Login to the third-party application and request the organization scope.

- Proceed through the interaction flow until reaching the consent page.

- An internal server error 500 is returned.

### Root Cause

For the /interaction/consent endpoint, the organization scope is returned alongside other resource scopes in the missingResourceScopes property.

In the consentInfoResponseGuard, we utilize the resource Scopes zod guard to validate the missingResourceScopes property. However, the description field in the resource scope is mandatory while the organization scopes' description is optional. An organization scope with an empty description will not pass the validation.

### Solution

Modify the consentInfoResponseGuard's missingResourceScopes property. Use the organization scope zod guard which does not necessitate the description field.

The alignment of the resource scope and organization scope types will be addressed in the next release.

## Testing

integration tests added

## Checklist

- [x] `.changeset`
- [ ] ~unit tests~
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: .changeset/seven-socks-perform.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+"@logto/schemas": patch
+---
+
+## Resolve Third-Party App's /interaction/consent Endpoint 500 Error
+
+### Reproduction Steps
```
Comment (gao-sun) [inline .changeset/seven-socks-perform.md:7]: can we use sentence case for titles?

File: .changeset/seven-socks-perform.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+- Login to the third-party application and request the organization scope.
+
+- Proceed through the interaction flow until reaching the consent page.
+
+- An internal server error 500 is returned.
+
+### Root Cause
+
+For the /interaction/consent endpoint, the organization scope is returned alongside other resource scopes in the missingResourceScopes property.
```
Comment (gao-sun) [inline .changeset/seven-socks-perform.md:19]: can we wrap code snippets with '`'?

File: .changeset/seven-socks-perform.md (hunk 1 above)
Comment (simeng-li) [reply .changeset/seven-socks-perform.md]: updated

File: packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/components/CreatePermissionModal/index.tsx [hunk 3]
```diff
@@ -118,10 +118,10 @@ function CreatePermissionModal({ resourceId, totalResourceCount, onClose }: Prop
               error={errors.name?.message}
             />
           </FormField>
-          <FormField isRequired title="api_resource_details.permission.description">
+          <FormField title="api_resource_details.permission.description">
```
Comment (gao-sun) [inline packages/console/src/pages/ApiResourceDetails/ApiResourcePermissions/components/CreatePermissionModal/index.tsx:121]: should we should `isRequired` here? is it by deisgn?

File: packages/experience/src/pages/Consent/ScopesListCard/index.tsx [hunk 4]
```diff
@@ -20,7 +21,7 @@ type ScopeGroupProps = {
   scopes: Array<{
     id: string;
     name: string;
-    description?: string;
+    description?: Nullable<string>; // Organization scopes description is nullable
```
Comment (gao-sun) [inline packages/experience/src/pages/Consent/ScopesListCard/index.tsx:24]:
```suggestion
    description?: Nullable<string>; // Organization scope description cloud be `null`
```

File: packages/integration-tests/src/client/index.ts [hunk 5]
```diff
@@ -84,7 +84,7 @@ export default class MockClient {
     assert(this.interactionCookie, new Error('Get cookie from authorization endpoint failed'));
   }
 
-  public async processSession(redirectTo: string) {
+  public async processSession(redirectTo: string, consent = true) {
```
Comment (gao-sun) [inline packages/integration-tests/src/client/index.ts:87]: better to add some comments about the parameters

**Review by gao-sun — APPROVED** [review]

File: packages/integration-tests/src/client/index.ts (hunk 5 above)
Comment (simeng-li) [reply packages/integration-tests/src/client/index.ts:92]: added

**Pull request merged**